### ISSUE TITLE
Exim.conf failregex

### ISSUE THREAD

**Issue by webappvicio** [fail2ban/fail2ban#2553]
LogFile line sample: 
```
Oct 22 03:39:17 mx1 exim[29786]: dovecot_login 	authenticator failed for (User) [92.118.38.37]: 535 Incorrect authentication data (set_id=takeuchi@domain.com)
```
doesn't trigger fail2bad using the filter.d/exim.conf(line 20) default failregex:
````
^%(pid)s \w+ authenticator failed for (?:[^\[\( ]* )?(?:\(\S*\) )?\[<HOST>\](?::\d+)?(?: I=\[\S+\](:\d+)?)?: 535 Incorrect authentication data( \(set_id=.*\)|: \d+ Time\(s\))?\s*$
````
I tested with fail2ban-regex, and changed the regexp, removing the trailing "^" like this:
````
 %(pid)s \w+ authenticator failed for (?:[^\[\( ]* )?(?:\(\S*\) )?\[<HOST>\](?::\d+)?(?: I=\[\S+\](:\d+)?)?: 535 Incorrect authentication data( \(set_id=.*\)|: \d+ Time\(s\))?\s*$
````
and it works.
I'm not sure if it is an issue on my log/fail2ban configuration or an issue to report. 
Thanks for any advice
vicio

on AWS amazon Linux2 EC2, failban 0.11.0.dev3, installed with 
```
cd /_install 
git clone https://github.com/fail2ban/fail2ban.git
cd fail2ban
sudo python setup.py install 
cp build/fail2ban.service /usr/lib/systemd/system/
systemctl start fail2ban
systemctl enable fail2ban
fail2ban-client version
```
Exim.conf enabled with jail.conf:
```
[exim]
logpath = /var/log/exim.log
enabled= true
```
/var/log/exim.log is created through rsyslog with output redirect:
```
if $programname == 'exim' then 	-/var/log/exim.log
& stop
```

[fail2ban-d.txt](https://github.com/fail2ban/fail2ban/files/3754823/fail2ban-d.txt)

**Comment by sebres** [comment]
Removing of anchor `^` is not a solution, rather interpolation of `pid` could be adjusted, because until now exim-filter does not expect prefix `mx1 exim[29786]:` in log.

So you could supply extended `pid` substitution (appending RE for this prefix):
```bash
fail2ban-client "$msg_or_log" 'exim[pid="(?: \w+ exim\[\d+\]:)?<known/pid>"]'
# fail2ban-client "$msg_or_log" 'exim[pid="(?: \w+ exim\[\d+\]:)?"]'
```
You could also remove here `<known/pid>` (interpolated as `(?: \[\d+\])?`), like in second example, if your log never contains ` [1234]` (starting with space) at begin of message (after the timestamp).

I'd extend it as default exim-common `pid` interpolation, just I'm unsure it is proper way, due to unknown rsyslog format to me, and because basically it is internal authentication from your dovecot (not exim directly), see below...

Anyway if you want to gather this as failures (unless it is not implemented in filter), you can provide it directly in your `jail.local`:
```ini
[exim]
filter = %(known/filter)s[pid="(?: \w+ exim\[\d+\]:)?<known/pid>"]
# filter = %(known/filter)s[pid="(?: \w+ exim\[\d+\]:)?"]
...
enabled = true
```
This should work.

And last but not least since it is **dovecot authenticator issue**, I'm pretty sure dovecot would log this authentication failure in own log too, so it is **advisable to use dovecot jail** for that (because they act basically on different ports and you'd ban it for dovecot not for exim).

**Comment by marcin-gryszkalis** [comment]
As I can see current failregex in dovecot.conf doesn't catch auth-worker failures (only those logged by imap-login process are caught). I didn't investigate it further as I prefer using exim log (as the original poster). In my case they look like 
```
Mar 17 11:48:12 auth-worker(80180): Info: conn unix:auth-worker (uid=143): auth-worker<13247>: sql(support,93.69.54.1): unknown user
Mar 06 23:06:05 auth-worker(57065): Info: conn unix:auth-worker (uid=143): auth-worker<225622>: sql(user@domain.com,18.6.8.4,<Yx7+W8+Io>): Password mismatch
```
(I don't expect it to be fixed unless somebody else requested the fix, I pasted it just for reference.

One note though: mismatching ports is exactly the problem you'll have if you use ip+port blocking, because in this case external connection comes to smtp port and blocking pop3/imap port won't have any effect - the default jail.conf doesn't list smtp in the list of ports for dovecot jail (I don't know why as smtps and submission are listed).

```
[dovecot]
port    = pop3,pop3s,imap,imaps,submission,465,sieve
```

**Comment by sebres** [comment]
Fixed in 8b984a0135be5ea284a2345cce0e3458dfb95d7b.

**Comment by sebres** [comment]
@marcin-gryszkalis regarding dovecot auth-worker messages: this kind of prefix is indeed unsupported - so the filter is unable to match part `conn unix:auth-worker (uid=143): auth-worker<13247>:`.
Since the fix for that is pretty simple (without a slowing down the filter) I've extended it in 92f90038fa67d719c55c75f5eff8059e849fe747. Thus it is also fixed now.
As for smpt-port for dovecot jail - yes it is not specified (no idea why)... although the ports are normally always matter of user configuration (at least user should verify it), dovecot is often used as an authenticator for exim and other smtp-services (which can listen smtp-port), so I tend to extend dovecot jail with that.
Thank you.
